src/overlay_12_0224E4FC.c

```diff
@@ -0,0 +1,1224 @@
+    
+    if (!CheckAbilityActive(bsys, ctx, 8, 0, ABILITY_CLOUD_NINE) && !CheckAbilityActive(bsys, ctx, 8, 0, ABILITY_AIR_LOCK)) {
+        if ((ability1 == ABILITY_SWIFT_SWIM && ctx->fieldCondition & FIELD_CONDITION_RAIN_ALL) ||
+            (ability1 == ABILITY_CHLOROPHYLL && ctx->fieldCondition & FIELD_CONDITION_SUN_ALL)) {
+                speed1 *= 2;
+            }        
+        if ((ability2 == ABILITY_SWIFT_SWIM && ctx->fieldCondition & FIELD_CONDITION_RAIN_ALL) ||
+            (ability2 == ABILITY_CHLOROPHYLL && ctx->fieldCondition & FIELD_CONDITION_SUN_ALL)) {
+                speed2 *= 2;
+            }
+    }
+    
+    for (i = 0; i < NELEMS(sSpeedHalvingItemEffects); i++) {
+        if (GetItemHoldEffect(ctx, ctx->battleMons[battlerId1].item, 1) == sSpeedHalvingItemEffects[i]) {
+            speed1 /= 2;
+            break;
+        }
+    }
+    
+    if (heldItem1 == HOLD_EFFECT_CHOICE_SPEED) {
+        speed1 = speed1 * 15/10;
+    }
+    
+    if (heldItem1 == HOLD_EFFECT_DITTO_SPEED_UP && ctx->battleMons[battlerId1].species == SPECIES_DITTO) {
+        speed1 *= 2;
+    }
+    
+    if (ability1 == ABILITY_QUICK_FEET && ctx->battleMons[battlerId1].status & 0xFF) {
+        speed1 = speed1 * 15/10;
+    } else if (ctx->battleMons[battlerId1].status & STATUS_PARALYSIS) {
+        speed1 /= 4;
+    }
+    
+    if (ability1 == ABILITY_SLOW_START && ctx->totalTurns - ctx->battleMons[battlerId1].unk88.slowStartTurns < 5) {
+        speed1 /= 2;
+    }
+    
+    if (ability1 == ABILITY_UNBURDEN && ctx->battleMons[battlerId1].unk88.knockOffFlag && ctx->battleMons[battlerId1].item == FALSE) {
+        speed1 *= 2;
+    }
+
+    if (ctx->fieldSideConditionFlags[BattleSys_GetFieldSide(bsys, battlerId1)] & SIDE_CONDITION_TAILWIND) {
+        speed1 *= 2;
+    }
+    
+    if (heldItem1 == HOLD_EFFECT_SOMETIMES_PRIORITY) {
+        if ((ctx->unk_310C[battlerId1])%(100/extra1) == 0) {
+            boostedPriority1 = 1;
+            
+            if (!flag) {
+                ctx->battleMons[battlerId1].unk88.quickClawFlag = TRUE;
+            }
+        }
+    }
+    
+    if (heldItem1 == HOLD_EFFECT_PINCH_PRIORITY) {
+        if (GetBattlerAbility(ctx, battlerId1) == ABILITY_GLUTTONY) {
+            extra1 /= 2;
+        }
+        if (ctx->battleMons[battlerId1].hp <= ctx->battleMons[battlerId1].maxHp/extra1) {
+            boostedPriority1 = 1;
+            if (!flag) {
+                ctx->battleMons[battlerId1].unk88.custapBerryFlag = TRUE;
+            }
+        }
+    }
+    
+    if (heldItem1 == HOLD_EFFECT_SPEED_DOWN) {
+        loweredPriority1 = 1;
+    }
+    
+    for (i = 0; i < NELEMS(sSpeedHalvingItemEffects); i++) {
+        if (GetItemHoldEffect(ctx, ctx->battleMons[battlerId2].item, 1) == sSpeedHalvingItemEffects[i]) {
+            speed2 /= 2;
+            break;
+        }
+    }
+    
+    if (heldItem2 == HOLD_EFFECT_CHOICE_SPEED) {
+        speed2 = speed2 * 15/10;
```

Comment:
space

Reply:
I think not having a space for smaller scale division looks better
`speed * 15 / 10` versus `speed * 15/10`
not super attached but that's my pref